k8s/README.md

```diff
@@ -0,0 +1,400 @@
++       line := text + " <3 k8s\n"
++       if _, err := stream.Write([]byte(line)); err != nil {
+                return err
+        }
+        doWrite := func() bool {
+-               _, err := stream.Write([]byte(" " + text))
++               _, err := stream.Write([]byte(line))
+                return err == nil
+        }
+        if limit == 0 {
+```
+
+#### Canary ####
+
+Then, I deploy my changes under new service and pod names:
+```
+:; kubectl apply -f k8s/gob/gen
+replicationcontroller "gen-growthhack" created
+service "gen-growthhack" created
+```
+
+Still, the site behaves as expected since I have not changed any
+routing policy:
+```
+:; curl -s "$GOB_HOST/gob?limit=10" 
+gob gob gob gob gob gob gob gob gob gob
+```
+
+Then, I'm able to test out the new service on a per-request basis by
+injecting a `Dtab-local` header with an override:
+
+```
+:; curl -H 'Dtab-local: /host/gen => /srv/gen-growthhack' "$GOB_HOST/gob?limit=10"
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+bees <3 k8s
+```
+
+This override says "whenever you refer to the _gen_ service, use
+_gen-growthhack_ instead.""
+
+#### Easing our growthhack into production ####
+
+For the sake of the demo, we need to generate some load on the site.
+To do so, the _reqz_ package can easily be launched as a one-off task
+with:
+
+```
+:; kubectl run --image=gcr.io/gobs-project/reqz:v0 reqz -- "http://$GOB_HOST/gob?limit=10&text=buoyant"
+deployment "reqz" created
+```
+
+Going to the admin page, we should see a few hundred requests per
+second on the site.
+
+Now we can update our dtab to send 1/20th of requests to the new
+service:
+```
+:; cat k8s/namerd/default.dtab 
+/srv        =>   /io.l5d.k8s/default/http;
+/host       =>   /srv;
+/method     =>   /$/io.buoyant.http.anyMethodPfx/host;
+/http/1.1   =>   /method;
+/host/gen   =>   1 * /srv/gen-growthhack & 20 * /srv/gen
+```
+
+```
+:; namectl update default k8s/namerd/default.dtab
+```
+
+And we see a small 5% of traffic go to the new service.
+
+By changing the last line to:
+
+```
+/host/gen   =>   1 * /srv/gen-growthhack & 5 * /srv/gen
+```
+
+and updating namerd:
+
+```
+:; namectl update default k8s/namerd/default.dtab
+```
+
+we now see 20% of the traffic going to the new service.  Everything's
+looking good.
+
+Next, we bring it to equal 50% with:
+```
+/host/gen   =>   /srv/gen-growthhack & /srv/gen
+```
+
+Finally, we're so happy with it, that we put all traffic onto the new
+service, but we still leave the old one around in case we have to kill
+the new one if something goes wrong:
+
+```
+/host/gen   =>   /srv/gen-growthhack | /srv/gen
+```
+```
+:; namectl update default k8s/namerd/default.dtab
+```
+
+Everything's looking good, until the CEO frantically comes running
+downstairs exclaiming "OUR CUSTOMERS HATE NEWLINES!!! ROLL IT BACK!".
+
+So, we quickly just delete the new service:
```

Comment:
Deleting the service is one option, updating the dtab to it's original state is another.  What are the pros v cons?
Maybe talk about the scenario where the new service has a bug and crashes hard?